app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java

```diff
@@ -124,9 +128,44 @@ public void run() {
         //todo: where to start service? login activity?
         //WakefulIntentService.sendWakefulWork(this, GGService.actionGetTipsIntent(this));
 
+        // Check if the GPS is enabled,
+        // and ask the user if he wants to enable it
+        checkIfGpsEnabled();
     }
 
-    private void CalculateCurrentLocation() {
+    private void checkIfGpsEnabled() {
+        final LocationManager manager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
```

Comment:
consider moving this method into LocationFetcher. this is where we use locationManager. maybe you can add there flag that will be updated by the LocationManager's receiver.